packages/arb-token-bridge-ui/src/hooks/useArbQueryParams.tsx

```diff
@@ -43,24 +43,35 @@ export const useArbQueryParams = () => {
   })
 }
 
+const isMax = (amount: string | undefined) =>
+  amount?.toLowerCase() === AmountQueryParamEnum.MAX
+
 // Our custom query param type for Amount field - will be parsed and returned as a string,
 // but we need to make sure that only valid numeric-string values are considered, else return '0'
 // Defined here so that components can directly rely on this for clean amount values and not rewrite parsing logic everywhere it gets used
 const AmountQueryParam = {
   // type of amount is always string | undefined coming from the input element onChange event `e.target.value`
-  encode: (amount: string | null | undefined) => amount,
+  encode: (amount: string | undefined = '') => {
+    const amountRegex = new RegExp(/^\d*(\.\d*)?$/)
+    if (isMax(amount)) {
+      return amount
+    }
+    return amountRegex.test(amount) ? amount : ''
+  },
   decode: (amount: string | (string | null)[] | null | undefined) => {
     const amountStr = amount?.toString()
 
+    if (isMax(amountStr)) {
+      return amountStr
+    }
+
     // to catch random string like `amount=asdf` from the URL
-    if (isNaN(Number(amountStr))) {
-      if (amountStr?.toLowerCase() === AmountQueryParamEnum.MAX) {
-        return amountStr
-      }
+    // to catch negative number
+    if (isNaN(Number(amountStr)) || Number(amountStr) < 0) {
       return ''
```

Comment:
Sorry, another comment. It seems like we have only one case when we want to return something else than `amountStr`, can we have it at the top, and return `amountStr` by default otherwise?

```suggestion
    // to catch random string like `amount=asdf` from the URL
    // to catch negative number
    if (isNaN(Number(amountStr)) || Number(amountStr) < 0) {
      return ''
    }
    
    return amountStr
```

Reply:
Discussed offline: it would not work as `max` would be caught by `if (isNaN(Number(amountStr))`